azurerm/internal/services/compute/virtual_machine_scale_set.go

```diff
@@ -1297,3 +1299,185 @@ func FlattenVirtualMachineScaleSetAutomaticRepairsPolicy(input *compute.Automati
 		},
 	}
 }
+
+func VirtualMachineScaleSetExtensionsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"name": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"publisher": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"type": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"type_handler_version": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"auto_upgrade_minor_version": {
+					Type:     schema.TypeBool,
+					Optional: true,
+					Default:  true,
```

Comment:
this could cause issues in some circumstances - should we make this Required to force users to consider it?

Reply:
This is the behaviour elsewhere in the provider, so I leaned towards consistency here.  That said I agree that the extension behaviour can be "unexpectedly different" between minor versions. The legacy VMSS resource also had this as optional, and the service default behaviour is also `true`, so I'm inclined to leave this as is?